### PR TITLE
Allow size calculations to be skipped

### PR DESCRIPTION
**Pull request by kjbracey-arm**
[X] I confirm this contribution is my own and I agree to license it with Apache 2.0.
[] I confirm the moderators may change the PR before merging it in.

### Summary of changes 

To minimise ROM space, add an option for constant, skipping the
size calculation.

### CODE REVIEW

**Comment by kjbracey-arm** [comment]
Fixed up unit test - I'd taken out the "do size calculation" call from the build thinking it was only needed for the `memset(0)`, but it also serves as a pre-sanity check for parameters.

(Which in turn means if the size calculation skip is enabled, there will be less parameter sanity checking.)

File: source/sn_coap_builder.c [hunk 1]
```diff
@@ -144,16 +142,26 @@ int16_t sn_coap_builder_2(uint8_t *dst_packet_data_ptr, const sn_coap_hdr_s *src
         /* * * * * * * * * * * * * * * * * * */
         sn_coap_builder_payload_build(&dst_packet_data_ptr, src_coap_msg_ptr);
     }
+
+    /* Shout as much as we can about overflow - if we exceed this, may have overrun user's buffer */
+    if (dst_packet_data_ptr - base_packet_data_ptr > dst_byte_count_to_be_built) {
```
Comment (AnttiKauppila) [inline source/sn_coap_builder.c:147]: Maybe we could add `tr_error` here to spot it from the traces?

Comment (kjbracey-arm) [reply source/sn_coap_builder.c:147]: I did want an assert, but the library doesn't already use them, so would be a new porting dependency. If this happens, we've probably overrun memory already - we don't get an "output buffer length" parameter.

A `tr_error` would make sense as a substitute, yes.

**Comment by kjbracey-arm** [comment]
This is showing include problems on the test job Tero set up - I'm not sure what the correct fix for the failing environment is.

(And we're getting run-time problems on the device it was intended for, but that could be straightforward lack of memory).

**Comment by kjbracey-arm** [comment]
Changed include in header from `mbed-coap/sn_config.h` to just `sn_config.h`, matching include style in other header, to try to resolve buid issues.

Added `tr_error` in overflow return.

**Review by TeroJaasko — APPROVED** [review]

**Pull request merged**